### PR TITLE
Move update offers call to RP

### PR DESCRIPTION
**Pull request by victor-meng**
There are currently two problems with RP's update offer call:
- RP's update database/collection throughput call has the same issue as the read throughput call where the response is missing properties like number of partitions and minimum throughput. I have to add an extra read database/collection offer call after the updating the offer.
- The migrate to auto pilot/manual call returns a "MethodNotAllowed" error even though the migration itself is successful. I have to add a try catch block to ignore the "MethodNotAllowed" errors.

### CODE REVIEW

File: src/Explorer/Controls/Settings/SettingsComponent.tsx [hunk 1]
```diff
@@ -426,7 +426,21 @@ export class SettingsComponent extends React.Component<SettingsComponentProps, S
             throw error;
           }
         } else {
-          const updatedOffer: DataModels.Offer = await updateOffer(this.collection.offer(), newOffer, headerOptions);
+          const updateOfferParams: DataModels.UpdateOfferParams = {
+            databaseId: this.collection.databaseId,
+            collectionId: this.collection.id(),
```
Comment (srinathnarayanan) [inline src/Explorer/Controls/Settings/SettingsComponent.tsx:431]: have you tested this using the feature flag "enableSettingsV2" ?
just to be sure, make sure you can update from autopilot -> manual and manual -> autopilot when you have the feature flag set

Comment (victor-meng) [reply src/Explorer/Controls/Settings/SettingsComponent.tsx:431]: Tested this change with the new settings tab and it worked fine.

**Review by southpolesteve — COMMENTED** [review]
One suggestion that I think is a typo

File: src/Common/dataAccess/readDatabaseOffer.ts [hunk 2]
```diff
@@ -75,24 +65,32 @@ const getDatabaseOfferIdWithARM = async (databaseId: string): Promise<string> =>
...
+        rpResponse = await getGremlinDatabaseThroughput(subscriptionId, resourceGroup, accountName, databaseId);
+        break;
+      default:
+        throw new Error(`Unsupported default experience type: ${defaultExperience}`);
+    }
+
+    return rpResponse?.name;
+  } catch (error) {
+    if (error.code !== "NotFound") {
+      throw new error();
```
[the remark this answers is not in the document]
Comment (victor-meng) [reply src/Common/dataAccess/readDatabaseOffer.ts]: Yep thanks for catching it!

**Review by southpolesteve — APPROVED** [review]

**Review by srinathnarayanan — APPROVED** [review]

**Pull request merged**